app/src/main/java/com/pajato/android/gamechat/exp/ChessFragment.java

```diff
@@ -669,37 +587,41 @@ private void handleTileBackground(final int index, final ImageButton currentTile
      * Locates the possible moves of the piece that is about to be highlighted.
      *
      * @param highlightedIndex the index containing the highlighted piece.
+     * @param possibleMoves a list of the possible moves available to the highlighted piece.
+     * @param board a HashMap representing an index on to board (0->63) the piece type at that location.
      */
-    private void findPossibleMoves(final int highlightedIndex, final ArrayList<Integer> possibleMoves) {
-        if(highlightedIndex < 0 || highlightedIndex > 64) {
+    private void findPossibleMoves(final int highlightedIndex, final ArrayList<Integer> possibleMoves,
```

Comment:
This method might be enhanced by returning a list of possible moves rather than clearing and updating the list.  Methods that return values via a switch statement are generally simpler and more concise than void methods.